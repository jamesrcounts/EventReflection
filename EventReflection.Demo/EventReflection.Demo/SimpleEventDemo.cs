using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace EventReflection.Demo
{
    [TestClass]
    public class SimpleEventDemo
    {
        [TestMethod]
        public void GetPocoEventInvocationList()
        {
            var poco = new Poco();
            poco.ProcessCompleted += Domain.HandleProcessCompleted;

            EventUtility.VerifyEventCallbacks(poco.GetEventHandlers());
        }

        [TestMethod]
        public void NullHasNoProcessCompletedHandler()
        {
            Assert.IsNull(ReflectionUtility.GetEventHandlers(null));
        }

        [TestMethod]
        public void PocoClientListensToPoco()
        {
            Poco poco = new Poco();
            PocoClient client = new PocoClient(poco);
            EventUtility.VerifyEventCallbacks(poco.GetEventHandlers());
        }

        [TestMethod]
        public void RaiseCompletedEventWhenProcessCompletes()
        {
            bool raisedEvent = false;
            Poco poco = new Poco();
            poco.ProcessCompleted += (s, e) => raisedEvent = true;
            poco.DoWork();
            Assert.IsTrue(raisedEvent);
        }
    }
}