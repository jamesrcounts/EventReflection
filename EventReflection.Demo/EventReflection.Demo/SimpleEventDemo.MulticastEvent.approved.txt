Event callbacks for Poco

ProcessCompleted
	[0] Void HandleProcessCompleted(System.Object, System.EventArgs)
	[1] Void HandleProcessStarted(System.Object, System.EventArgs)

